### PR TITLE
[WIP] Libwork

### PR DESCRIPTION
**Pull request by updega2**
[WIP] Set up for shared parasolid library on cmake

### CODE REVIEW

File: Code/Source/UI/gdscInit.cxx [hunk 1]
```diff
@@ -287,7 +287,7 @@ int SimVascular_Init( Tcl_Interp *interp )
 #endif
 
 #ifdef USE_PARASOLID
-#ifndef USE_PARASOLID_SHARED
+#if USE_PARASOLID_SHARED == 0
```
Comment (jmerkow) [inline Code/Source/UI/gdscInit.cxx]: I think this is what you were asking me about today.  You can use ifdef, this variable will not be defined with the cmake define. #cmakedefine SOME_DEFINE @SOME_DEFINE@ will become /* #undef SOME_DEFINE */ if not defined.

Here is the ZLIB def that is defined in the same file:
https://github.com/SimVascular/SimVascular/blob/b730da692c3c69d716fc7b0153f2327b5f76e548/Code/Source/PostProcessing/cvConvertVisFiles.h#L46

**Comment by jmerkow** [comment]
Which Meshsim/parasolid versions should I test this for?

**Comment by updega2** [comment]
I was using meshsim 9.0 and parasolid 26.1

**Comment by osmsc** [comment]
You need to add -fPIC to the compiler options for VTK when you compile the static libraries so that you can link against them when creating shared objects for SV.

I'm using Parasolid 26.1 and Meshsim 9.0 from July 4, 2005.

**Comment by updega2** [comment]
Oh yea, sorry, I used -fPIC to compile, but forgot to add it to the cmake flags when building with shared.

**Comment by jmerkow** [comment]
@updega2 So re-compile vtk with fPIC  and link againist parasolid 26.1 and meshsim 9?
Anything else I need to set?

**Comment by osmsc** [comment]
I just pushed an update to the ParasolidSolidModel code on github last night that has the required DLL_EXPORT functions for parasolid to link as shared.  So you should update to the latest master version of that as well.

**Comment by updega2** [comment]
Yea, there are a couple things that need to be changed in the parasolid cmake file that need to be changed as well. I can do that. 

Adam

**Comment by jmerkow** [comment]
@updedga2 , do you want to update the submodule hash in the main repo then rebase this PR?

**Comment by updega2** [comment]
Sure, but I won't be able to do it until later today. 

Adam

**Pull request merged**